pkg/merger/merger_test.go

```diff
@@ -408,6 +408,71 @@ func Test_MergeAndUpdate(t *testing.T) {
 		})
 	}
 }
+func Test_RecordLastModified(t *testing.T) {
+	cases := []struct {
+		name   string
+		attrs  *storage.ReaderObjectAttrs
+		mets   *Metrics
+		source string
+	}{
+		{
+			name:   "nil attrs; succeeds",
+			attrs:  nil,
+			mets:   &Metrics{},
+			source: "fake-source",
+		},
+		{
+			name:   "nil mets; succeeds",
+			attrs:  &storage.ReaderObjectAttrs{},
+			mets:   nil,
+			source: "fake-source",
+		},
+		{
+			name:   "nil; succeeds",
+			attrs:  nil,
+			mets:   nil,
+			source: "",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			recordLastModified(tc.attrs, tc.mets, tc.source)
+		})
+	}
+}
+
+func Test_RecordLastModified_RecordMetrics(t *testing.T) {
+	var fakeMetric FakeInt64
+
+	attrs := &storage.ReaderObjectAttrs{
+		LastModified: time.Now(),
+	}
+	mets := &Metrics{
+		LastModified: &fakeMetric,
+	}
+	source := ""
+
+	recordLastModified(attrs, mets, source)
+	if fakeMetric.last != 0 {
```

Comment:
This test relies on the test taking less than one second to execute. Probably a safe bet, but there might be a better way to check that the metric is calculated correctly (and not just a golang zero-value)

```
LastModified: time.Now().Sub(5 * time.Second)
...
if fakeMetric.last < tc.ExpectAtLeast {
     t.Errorf("want at least %d, got %d", tc.ExpectAtLeast, fakeMetric.last)
}
```